qiskit/circuit/delay.py

```diff
@@ -72,7 +72,10 @@ def validate_parameter(self, parameter):
             return parameter
         elif isinstance(parameter, float):
             if self.unit == "dt":
-                raise CircuitError("Integer duration is expected for 'dt' unit.")
+                parameter_int = int(parameter)
```

Comment:
Is this needed for loading from QASM? (Does the QASM parser parse all numbers a floats?)

Reply:
QASM parser converts all the non-qubits paramters into floats, yes.